docker/README.md

```diff
@@ -0,0 +1,168 @@
+
+```bash
+$ docker images
+REPOSITORY    TAG       IMAGE ID       CREATED          SIZE
+kubestellar   v0.3.1    0425aee2cda6   8 minutes ago    617MB
+kubestellar   v0.3.0    76d714261c6e   14 minutes ago   617MB
+```
+
+## Build and push a multi-architecture image
+
+Note that `docker buildx` is required for this option.
+
+Make sure to login into the registry with the desired user:
+
+```bash
+docker login -u <user> <registry>
+```
+
+Build and push a multi-architecture image using the command:
+
+```bash
+make buildx IMG=<registry>/<organization>/<image>
+```
+
+Several arguments can be used to customize the build:
+
+- `KUBESTELLAR_VERSION=<version>` is used to determine the version of **KubeStellar** being installed in the container image, *e.g.* `v0.3.1`. In the argument is not specified, it will default to the **stable** **KubeStellar** release version specified in https://github.com/kubestellar/kubestellar/blob/main/VERSION.
+- `TAG=<tag>` is used to determine the tag of the container image, *e.g.* `stable`. In the argument is not specified, it will default to the value set by `KUBESTELLAR_VERSION`.
+- `PLATFORMS` lists the desired target platforms. Default value; `PLATFORMS=linux/amd64,linux/arm64,linux/ppc64le`. It should be noted that a `linux/s390x` **KubeStellar**  cannot be build because **kcp** does not have a `linux/s390x` binary release.
+
+As an example, the **KubeStellar** container images available at https://quay.io/repository/kubestellar/kubestellar?tab=tags have been built with commands like:
+
+```bash
+make buildx IMG=quay.io/kubestellar/kubestellar KUBESTELLAR_VERSION=v0.4.0
+```
+
+## Run the container and access KubeStellar from the host OS
+
+Spin up a local or remote container image with the command:
+
+```bash
+make run
+export KUBECONFIG=${HOME}/.kcp/admin.kubeconfig
+export PATH=$PATH:${HOME}/kcp:${HOME}/kubestellar/bin
+```
+
+The following arguments can be used:
+
+- `IMG=[<registry>/<organization>/]<image>` select the container image, the default value is the local `kubestellar` image. Use `IMG=quay.io/kubestellar/kubestellar` to pull the **stable** container image from our [quay.io registry](https://quay.io/repository/kubestellar/kubestellar?tab=tags).
+- `TAG=<tag>`, select the image tag, the default value is the **stable** version of **KubeStellar** as specified in https://github.com/kubestellar/kubestellar/blob/main/VERSION.
+- `NAME=<container-name>` sets the name of the container, the default value is `kubestellar`
+- `BASEPATH=<path>` set the base path where the **KUBECONFIG** and executable will be made available. By default, `~` is used. Note that this parameter will impact the exact path of the `export` commands above, which are provided for the default case. The `make run` command will print out the correct export commands to be used.
+
+The command above will:
+
+- share a `${BASEPATH}/.kcp` folder containing the `KUBECONFIG` file

```

Comment:
(1) "share" is a little strange from a user's point of view, this `make run` does not cause the user to share that directory with anyone else.  I suppose that you are talking about sharing with the stuff inside the container, but from the user's point of view that is not important (unless they are also a contributor who cares about what goes on inside the container, but that is of secondary importance here).  What a user needs to know is that this command creates the directory if it does not already exist and that the kubeconfig file will appear in it.

(2) The typography of `KUBECONFIG` is potentially confusing, that could be read as a literal filename.  I would suggest writing `$KUBECONFIG` or `admin.kubeconfig` instead.

Reply:
@francostellari you can simply replace 'share' with 'mount on the local filesystem' and I would also use "admin.kubeconfig" to remain consistent with other instruction sets we have.  Just make these 2 changes to move past this item.

Reply:
"mount" is a dyadic operator: it makes one thing appear in another location.  Being explicit about both is important because not everyone will know what you are assuming about the thing not mentioned.

Reply:
Also, it is jarring that the last two entries in this list lead with creating a folder and the first two do not --- but in fact all four items in this list create a folder (if it does not already exist) on the host and this is more important to a user than anything that happens inside the container.

Reply:
> @francostellari you can simply replace 'share' with 'mount on the local filesystem' and I would also use "admin.kubeconfig" to remain consistent with other instruction sets we have. Just make these 2 changes to move past this item.

1) I have changed `share` to `mount on the host filesystem`
2) I have changed `KUBECONFIG` to `admin-kubeconfig`